Test failure

Windows test often fail with this:
```
    CompositeException (5 nested):
    -------------------------------------------
    [0]
    java.lang.AssertionError: unbalanced `beforeRootsChange`/`rootsChanged`: -1
    [1]
    java.lang.AssertionError: unbalanced `beforeRootsChange`/`rootsChanged`: -1
    [2]
    java.lang.AssertionError: unbalanced `beforeRootsChange`/`rootsChanged`: -1
    [3]
    java.lang.AssertionError: unbalanced `beforeRootsChange`/`rootsChanged`: -1
    [4]
    java.lang.AssertionError: unbalanced `beforeRootsChange`/`rootsChanged`: -1
```

Update to 2023.2.1 should fix this.

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -14,7 +14,7 @@ jobs:
     runs-on: ${{ matrix.image }}
     strategy:
       matrix:
-        image: [ubuntu-22.04, windows-2022] # TODO: [macos-12, ubuntu-22.04, windows-2022] Tests are broken on macOS, uncomment in EAP2
+        image: [macos-12, ubuntu-22.04, windows-2022]
       fail-fast: false
     env:
       DOTNET_NOLOGO: 1

--- a/gradle.properties
+++ b/gradle.properties
@@ -3,7 +3,7 @@ kotlin.stdlib.default.dependency=false
 pluginVersionBase=1.1.0
 
 # https://www.jetbrains.com/intellij-repository/snapshots/
-riderSdkVersion=2023.2
+riderSdkVersion=2023.2.1
 untilBuildVersion=233.*
 
 buildRelease=false

--- a/src/test/kotlin/me/fornever/avaloniarider/test/cases/MsBuildParameterCollectorTests.kt
+++ b/src/test/kotlin/me/fornever/avaloniarider/test/cases/MsBuildParameterCollectorTests.kt
@@ -14,6 +14,7 @@ import com.jetbrains.rider.test.env.enums.SdkVersion
 import me.fornever.avaloniarider.model.RdProjectOutput
 import me.fornever.avaloniarider.previewer.MsBuildParameterCollector
 import me.fornever.avaloniarider.test.framework.AvaloniaIntegrationTest
+import me.fornever.avaloniarider.test.framework.correctTestSolutionDirectory
 import me.fornever.avaloniarider.test.framework.runPumping
 import org.testng.annotations.Test
 
@@ -31,7 +32,7 @@ class MsBuildParameterCollectorTests : AvaloniaIntegrationTest() {
         runnableProjects.shouldContains { it.kind == RunnableProjectKinds.DotNetCore }
         runnableProjects.shouldContains { it.kind == RunnableProjectKinds.LaunchSettings }
 
-        val projectFilePath = activeSolutionDirectory.resolve("MSBuildParameters.csproj").toPath()
+        val projectFilePath = correctTestSolutionDirectory.resolve("MSBuildParameters.csproj")
         val projectOutput = runnableProjects.single { it.kind == RunnableProjectKinds.DotNetCore }
             .projectOutputs.single()
         val projectModelEntity = workspaceModel.getProjectModelEntities(projectFilePath, project).single()
@@ -43,7 +44,7 @@ class MsBuildParameterCollectorTests : AvaloniaIntegrationTest() {
             )
         }
 
-        val previewerPathForTest = activeSolutionDirectory.resolve("PreviewerForTest.exe").toPath()
+        val previewerPathForTest = correctTestSolutionDirectory.resolve("PreviewerForTest.exe")
         parameters.previewerBinary.shouldBe(previewerPathForTest)
     }
 }

--- a/src/test/kotlin/me/fornever/avaloniarider/test/cases/MsBuildParameterCollectorTests.kt
+++ b/src/test/kotlin/me/fornever/avaloniarider/test/cases/MsBuildParameterCollectorTests.kt
@@ -14,7 +14,6 @@ import com.jetbrains.rider.test.env.enums.SdkVersion
 import me.fornever.avaloniarider.model.RdProjectOutput
 import me.fornever.avaloniarider.previewer.MsBuildParameterCollector
 import me.fornever.avaloniarider.test.framework.AvaloniaIntegrationTest
-import me.fornever.avaloniarider.test.framework.canonicalSolutionDirectory
 import me.fornever.avaloniarider.test.framework.runPumping
 import org.testng.annotations.Test
 
@@ -32,7 +31,7 @@ class MsBuildParameterCollectorTests : AvaloniaIntegrationTest() {
         runnableProjects.shouldContains { it.kind == RunnableProjectKinds.DotNetCore }
         runnableProjects.shouldContains { it.kind == RunnableProjectKinds.LaunchSettings }
 
-        val projectFilePath = canonicalSolutionDirectory.resolve("MSBuildParameters.csproj")
+        val projectFilePath = activeSolutionDirectory.resolve("MSBuildParameters.csproj").toPath()
         val projectOutput = runnableProjects.single { it.kind == RunnableProjectKinds.DotNetCore }
             .projectOutputs.single()
         val projectModelEntity = workspaceModel.getProjectModelEntities(projectFilePath, project).single()
@@ -44,7 +43,7 @@ class MsBuildParameterCollectorTests : AvaloniaIntegrationTest() {
             )
         }
 
-        val previewerPathForTest = canonicalSolutionDirectory.resolve("PreviewerForTest.exe")
+        val previewerPathForTest = activeSolutionDirectory.resolve("PreviewerForTest.exe").toPath()
         parameters.previewerBinary.shouldBe(previewerPathForTest)
     }
 }

--- a/src/test/kotlin/me/fornever/avaloniarider/test/cases/PreviewTests.kt
+++ b/src/test/kotlin/me/fornever/avaloniarider/test/cases/PreviewTests.kt
@@ -16,6 +16,7 @@ import com.jetbrains.rider.xaml.core.XamlPreviewEditorExtension
 import me.fornever.avaloniarider.controlmessages.FrameMessage
 import me.fornever.avaloniarider.idea.editor.AvaloniaPreviewerXamlEditorExtension
 import me.fornever.avaloniarider.previewer.AvaloniaPreviewerSessionController
+import me.fornever.avaloniarider.test.framework.correctTestSolutionDirectory
 import org.testng.annotations.Test
 import java.time.Duration
 
@@ -27,10 +28,10 @@ class PreviewTests : BaseTestWithSolution() {
     override val traceScenarios = setOf(RdLogTraceScenarios.Commands)
 
     private val mainWindowFile
-        get() = getVirtualFileFromPath("Views/MainWindow.xaml", activeSolutionDirectory)
+        get() = getVirtualFileFromPath("Views/MainWindow.xaml", correctTestSolutionDirectory.toFile())
 
     private val projectFilePathProperty
-        get() = OptProperty(activeSolutionDirectory.resolve("AvaloniaMvvm.csproj").toPath())
+        get() = OptProperty(correctTestSolutionDirectory.resolve("AvaloniaMvvm.csproj"))
 
     @Test
     fun previewEditorProviderShouldHandleTheXamlFile() {

--- a/src/test/kotlin/me/fornever/avaloniarider/test/cases/PreviewTests.kt
+++ b/src/test/kotlin/me/fornever/avaloniarider/test/cases/PreviewTests.kt
@@ -16,7 +16,6 @@ import com.jetbrains.rider.xaml.core.XamlPreviewEditorExtension
 import me.fornever.avaloniarider.controlmessages.FrameMessage
 import me.fornever.avaloniarider.idea.editor.AvaloniaPreviewerXamlEditorExtension
 import me.fornever.avaloniarider.previewer.AvaloniaPreviewerSessionController
-import me.fornever.avaloniarider.test.framework.canonicalSolutionDirectory
 import org.testng.annotations.Test
 import java.time.Duration
 
@@ -28,10 +27,10 @@ class PreviewTests : BaseTestWithSolution() {
     override val traceScenarios = setOf(RdLogTraceScenarios.Commands)
 
     private val mainWindowFile
-        get() = getVirtualFileFromPath("Views/MainWindow.xaml", canonicalSolutionDirectory.toFile())
+        get() = getVirtualFileFromPath("Views/MainWindow.xaml", activeSolutionDirectory)
 
     private val projectFilePathProperty
-        get() = OptProperty(canonicalSolutionDirectory.resolve("AvaloniaMvvm.csproj"))
+        get() = OptProperty(activeSolutionDirectory.resolve("AvaloniaMvvm.csproj").toPath())
 
     @Test
     fun previewEditorProviderShouldHandleTheXamlFile() {

--- a/src/test/kotlin/me/fornever/avaloniarider/test/cases/RunnableAssemblySelectorActionTests.kt
+++ b/src/test/kotlin/me/fornever/avaloniarider/test/cases/RunnableAssemblySelectorActionTests.kt
@@ -27,6 +27,7 @@ import me.fornever.avaloniarider.idea.settings.AvaloniaProjectSettings
 import me.fornever.avaloniarider.idea.settings.AvaloniaWorkspaceSettings
 import me.fornever.avaloniarider.model.avaloniaRiderProjectModel
 import me.fornever.avaloniarider.test.framework.AvaloniaIntegrationTest
+import me.fornever.avaloniarider.test.framework.correctTestSolutionDirectory
 import org.testng.Assert.assertFalse
 import org.testng.annotations.AfterMethod
 import org.testng.annotations.BeforeMethod
@@ -52,7 +53,7 @@ class RunnableAssemblySelectorActionTests : AvaloniaIntegrationTest() {
     }
 
     private val testXamlFile
-        get() = getVirtualFileFromPath("ClassLibrary1/MyControl.axaml", activeSolutionDirectory)
+        get() = getVirtualFileFromPath("ClassLibrary1/MyControl.axaml", correctTestSolutionDirectory.toFile())
 
     private fun createMockSelector(
         isSolutionLoading: IOptPropertyView<Boolean> = OptProperty(),
@@ -148,7 +149,7 @@ class RunnableAssemblySelectorActionTests : AvaloniaIntegrationTest() {
         val action = RunnableAssemblySelectorAction(
             testLifetime,
             project,
-            getVirtualFileFromPath("Views/MainWindow.xaml", activeSolutionDirectory)
+            getVirtualFileFromPath("Views/MainWindow.xaml", correctTestSolutionDirectory.toFile())
         )
         pumpMessages(Duration.ofSeconds(5L)) { !action.isLoading.value }.shouldBeTrue()
 

--- a/src/test/kotlin/me/fornever/avaloniarider/test/cases/RunnableAssemblySelectorActionTests.kt
+++ b/src/test/kotlin/me/fornever/avaloniarider/test/cases/RunnableAssemblySelectorActionTests.kt
@@ -27,7 +27,6 @@ import me.fornever.avaloniarider.idea.settings.AvaloniaProjectSettings
 import me.fornever.avaloniarider.idea.settings.AvaloniaWorkspaceSettings
 import me.fornever.avaloniarider.model.avaloniaRiderProjectModel
 import me.fornever.avaloniarider.test.framework.AvaloniaIntegrationTest
-import me.fornever.avaloniarider.test.framework.canonicalSolutionDirectory
 import org.testng.Assert.assertFalse
 import org.testng.annotations.AfterMethod
 import org.testng.annotations.BeforeMethod
@@ -53,7 +52,7 @@ class RunnableAssemblySelectorActionTests : AvaloniaIntegrationTest() {
     }
 
     private val testXamlFile
-        get() = getVirtualFileFromPath("ClassLibrary1/MyControl.axaml", canonicalSolutionDirectory.toFile())
+        get() = getVirtualFileFromPath("ClassLibrary1/MyControl.axaml", activeSolutionDirectory)
 
     private fun createMockSelector(
         isSolutionLoading: IOptPropertyView<Boolean> = OptProperty(),
@@ -149,7 +148,7 @@ class RunnableAssemblySelectorActionTests : AvaloniaIntegrationTest() {
         val action = RunnableAssemblySelectorAction(
             testLifetime,
             project,
-            getVirtualFileFromPath("Views/MainWindow.xaml", canonicalSolutionDirectory.toFile())
+            getVirtualFileFromPath("Views/MainWindow.xaml", activeSolutionDirectory)
         )
         pumpMessages(Duration.ofSeconds(5L)) { !action.isLoading.value }.shouldBeTrue()
 

--- a/src/test/kotlin/me/fornever/avaloniarider/test/framework/AvaloniaIntegrationTest.kt
+++ b/src/test/kotlin/me/fornever/avaloniarider/test/framework/AvaloniaIntegrationTest.kt
@@ -1,6 +1,7 @@
 package me.fornever.avaloniarider.test.framework
 
 import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.SystemInfo
 import com.jetbrains.rider.test.OpenSolutionParams
 import com.jetbrains.rider.test.base.BaseTestWithSolution
 import java.io.File
@@ -16,5 +17,13 @@ abstract class AvaloniaIntegrationTest : BaseTestWithSolution() {
     }
 }
 
-val BaseTestWithSolution.canonicalSolutionDirectory: Path
-    get() = activeSolutionDirectory.canonicalFile.toPath()
+/**
+ * On Windows, it has to resolve the canonical path (presumably to resolve short paths), but that could break tests on
+ * macOS, so we disable canonical paths on that.
+ */
+val BaseTestWithSolution.correctTestSolutionDirectory: Path
+    get() =
+        if (SystemInfo.isWindows)
+            activeSolutionDirectory.canonicalFile.toPath()
+        else
+            activeSolutionDirectory.toPath()

